### PR TITLE
Fix deadlock in `recordAd` and `LaunchTTLCache` and add support to turn off `stat`

### LINKED ISSUE #1438: Deadlock in the director between `statUtilsMutex` and `healthTestUtilsMutex`
Pelican Version:
_Run `./pelican --version` to get version number._

Pelican Service:
- [ ] Director

**Describe the bug**
_A clear and concise description of what the bug is._

We flipped the order to locking when acquiring both `statUtilsMutex` and `healthTestUtilsMutex` in `recordAd` and `LaunchTTLCache`, causing the deadlock.

**To Reproduce**
_Steps to reproduce the behavior._

It's difficult to reproduce the issue as multiple attempts in ITB and locally failed. But the code is relatively straightforward in suggesting the deadlock condition

**Expected behavior**
_A clear and concise description of what you expected to happen._

**Screenshots**
_If applicable, add screenshots to help explain your problem._

**Desktop (please complete the following information):**
 - OS: [e.g. Linux, MacOS]
 - Browser: [e.g. chrome, safari, if the bug relates to any web user interface]

**Additional context**
_Add any other context about the problem here._

### LINKED ISSUE #1440: Add config knob to turn off `stat` in the director

### PR DESCRIPTION
**Pull request by haoming29**
Fixes #1438 and closes #1440

### CODE REVIEW

File: director/director.go [hunk 1]
```diff
@@ -397,8 +397,15 @@ func redirectToOrigin(ginCtx *gin.Context) {
 	reqPath := path.Clean("/" + ginCtx.Request.URL.Path)
 	reqPath = strings.TrimPrefix(reqPath, "/api/v1.0/director/origin")
 
+	disableStat := param.Director_DisableStat.GetBool()
```
Comment (jhiemstrawisc) [inline director/director.go:400]: Can we change the config name to match the style we use elsewhere, i.e. `Director.EnableStat` and then flip the default to be `true`?

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]

File: director/director.go (hunk 1 above)
Comment (haoming29) [reply director/director.go:400]: I thought the idea/guideline is to name the config in a way that the default value is `false`? No?

Comment (jhiemstrawisc) [reply director/director.go:400]: Hmm, I don't know that we've settled on a formal guideline, have we? Looking over `docs/parameters.yaml` shows that we're mostly using `.Enable*`-type parameters (some of which default to true, and some to false), although we do have one "Disable" param that doesn't come from a legacy StashCP configuration (`Logging.DisableProgressBars`). If you prefer the param as is, I'll approve and let you make the decision.

**Review by jhiemstrawisc — APPROVED** [review]

File: director/director.go (hunk 1 above)
Comment (haoming29) [reply director/director.go:400]: I'll do `Enable*` then just to keep things consistent

**Comment by haoming29** [comment]
@jhiemstrawisc I've updated to use `EnableStat` instead. Not sure if you want to take another look

**Pull request merged**